src/vs/platform/terminal/common/capabilities/commandDetection/promptInputModel.ts

```diff
@@ -456,6 +456,9 @@ export class PromptInputModel extends Disposable implements IPromptInputModel {
 			ghostTextIndex = this._scanForGhostTextAdvanced(buffer, line, cursorIndex);
 		}
 
+		if (ghostTextIndex > -1 && this.value.substring(ghostTextIndex).endsWith(' ')) {
+			this._value = this.value.trim();
+		}
```

Comment:
@meganrogge do we need to re-evaluate here whether `ghostTextIndex` is still valid after the trim? For example something like this: `foo |[   ]`.